components/style/rule_tree/mod.rs

```diff
@@ -580,13 +586,13 @@ impl StrongRuleNode {
                 let existing =
                     next_sibling_ptr.compare_and_swap(ptr::null_mut(),
                                                       new_ptr,
-                                                      Ordering::SeqCst);
+                                                      Ordering::Acquire);
```

Comment:
Hm, doesn't this need to be AcqRel? It needs to be Release because of the issue I pointed out last time (same reason why next_sibling() needs to use Acquire()). I think it needs to be Acquire if something reads fields on |existing|, which happens in the upgrade() call below.